### PR TITLE
WaitFor for Cosmos DB

### PR DESCRIPTION
**Pull request by mitchdenny**
## Description

This PR adds `WaitFor` support for Cosmos DB.

Related #5645 

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
  - [ ] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Link to aspire-docs issue: 
  - [x] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5729)

### CODE REVIEW

**Comment by mitchdenny** [comment]
This is ready. I held it in draft until I understood the issue with `WithLifetime(...)` not working the way I expected, but that is this issue: #5741

File: playground/CosmosEndToEnd/CosmosEndToEnd.AppHost/Program.cs [hunk 1]
```diff
@@ -3,13 +3,21 @@
 
 var builder = DistributedApplication.CreateBuilder(args);
 
+#pragma warning disable ASPIRECONTAINERLIFETIME001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
 var db = builder.AddAzureCosmosDB("cosmos")
                 .AddDatabase("db")
-                .RunAsEmulator();
+                .RunAsEmulator(c => {
+                    c.WithLifetime(ContainerLifetime.Persistent);
```
Comment (davidfowl) [inline playground/CosmosEndToEnd/CosmosEndToEnd.AppHost/Program.cs:10]: undo

File: src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBExtensions.cs [hunk 2]
```diff
@@ -69,9 +73,45 @@ public static IResourceBuilder<AzureCosmosDBResource> AddAzureCosmosDB(this IDis
...
+            if (connectionString == null)
+            {
+                throw new DistributedApplicationException($"ConnectionStringAvailableEvent was published for the '{resource.Name}' resource but the connection string was null.");
+            }
+        });
+
+        var healthCheckKey = $"{name}_check";
+        builder.Services.AddHealthChecks().AddAzureCosmosDB(sp =>
+        {
+            var clientOptions = new CosmosClientOptions();
```
Comment (davidfowl) [inline src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBExtensions.cs:92]: Does this get called once? If no, we should move this to after the connection string resolution and cache the client  like in the Milvus PR

**Review by davidfowl — APPROVED** [review]

**Pull request merged**